### ISSUE TITLE
False-positive RedundantCondition with null-coalesce

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#659]
```php
<?php

/**
 * @return array<string,bool>
 */
function getOptions(): array
{
    return [
        'option' => true,
    ]
}

if (rand(0,1)) {
    $options = getOptions();
}

$option = $options['option'] ?? false;

if ($option) {
    // do something
}
```
Here, `$option` could be both `true` and `false`, but psalm thinks `if ($option)` is redundant:
```
ERROR: RedundantCondition - q.php:19:5 - Found a redundant condition when evaluating $option and trying to reconcile type '' to !falsy
if ($option) {
```

**Comment by weirdan** [comment]
It appears offsets on possibly-undefined arrays are treated as certainly undefined.

**Comment by muglug** [comment]
This is a consequence of https://github.com/vimeo/psalm/commit/60c6bb40218c9e26aec15347c4403c6f58f84328

**Comment by muglug** [comment]
You can fix in the meantime by setting `$options = []` above the `if`.

**Comment by weirdan** [comment]
That's what I did. In real code that `if` was depending on a method call on a mixed value, so I had to type that first.

**Comment by muglug** [comment]
Fixed in 2b9a2d3
